### ISSUE TITLE
video.add_text never updates in version 2.0.1+

### ISSUE THREAD

**Issue by vaguelyr** [savonet/liquidsoap#2302]
**Describe the bug**
video.add_text gets the text to display, but never updates, even if its getting a reference that is changed. It works on 2.0.0 but not after 2.0.1
I opened a discussion wondering if I was using it wrong but now I think it's a bug. https://github.com/savonet/liquidsoap/discussions/2301

**To Reproduce**
https://github.com/savonet/liquidsoap/discussions/2301
This script has text that is updated every second to create a clock overlay. 
It works on 2.0.0 but not on 2.0.1 or later versions.
```
myvideo = single("zoop.jpg")
radio = input.ffmpeg("https://xxxxx/main.mp3")
radio = mux_video(video=myvideo, radio)
radio = mksafe(radio)

# output
# periodically updating clock overlay ?
srightnow = ref("<time1>")
def update_time()
        print("updt from " ^ !srightnow)
        srightnow := process.read("date | tr -d '\n'")
        print("updt now  " ^ !srightnow)
end
thread.run(every=1.,update_time)

def gettime() = !srightnow end
radio = video.add_text( color=0x9999ff, speed=0, x=50, y=50, size=50, gettime, radio)

# output        
ykey = string.trim(file.contents("../streamkey"))
url = "rtmp://192.168.1.122:1935/live/#{ykey}"
enc = %ffmpeg(format="flv",
        %audio(codec="aac"),
        %video(codec="libx264"))
output.url(fallible=true, url=url, enc, radio)
```

**Expected behavior**
The video overlay text should change every second, creating a clock. 
It works on 2.0.0 and not after 2.0.1

**Version details**
ubuntu 20.04 LTS
tested with liquidsoap 2.0.0, 2.0.1-3

**Install method**
download from github releases > 
`sudo apt install ./liquidsoap_2.0.0-ubuntu-focal-1_amd64.deb 
`

**Comment by smimram** [comment]
I will have a closer look at your script, but a simple test script seems to indicate that `video.add_text` does indeed update the text for instance, I do see a counter with
```
s = blank()
s = video.add_text({string_of(time.up())}, s)
output.video(s)
```

**Comment by smimram** [comment]
I can confirm that your test is behaving as expected on the latest git here...

**Comment by toots** [comment]
I think some logs would help, `video.add_text` is picked up depending on the enabled modules. It was switched back to default to sdl first on Jan. 31 so for `2.0.3`

**Comment by vaguelyr** [comment]
Thanks for taking a look at it. I'll try installing on a fresh vm and see if that helps. 

Here are the logs, ran with 'liquidsoap try3.liq  > logs.txt 2>&1' and 
settings.log.level.set(5)
[logs.txt](https://github.com/savonet/liquidsoap/files/8319636/logs.txt)

**Comment by vaguelyr** [comment]
I installed a fresh ubuntu 20.04 lts and it has the same issue. 
Here's the log from the same script ran the same way.
[logs.txt](https://github.com/savonet/liquidsoap/files/8319888/logs.txt)

**Comment by toots** [comment]
Thanks for checking on this. I think I found the issue. `video.add_text.ffmpeg` was switched to a scripted value in `v2.0.1` but the switch did not actually register as top-level `video.add_text` so anyone using `video.add_text` previously via `ffmpeg` was actually switched to a different implementation.

I have just pushed a fix that:
* Makes sure `video.add_text` actually defaults to `ffmpeg` as expected.
* Adds `settings.video.add_text` to track which implementation is used at runtime for `video.add_text`, log it at start and issue a warning if the expected implementation is not available.

I'd suggest to add:
```ruby
settings.video.add_text.set("ffmpeg")
```
It you have tested and are expecting the `"ffmpeg"` implementation for this operator. Change to whichever implementation you expect.

Sidenote: I am personally not a super fan of switching implementation without telling the user but it's pattern that seems useful for new users when you do not want to have to know about underlying implementations. We already do that for audio input and output.
